### DISCUSSION TITLE
null check for operator[] for const json object

### DISCUSSION THREAD

**Discussion by Wokarol** [nlohmann/json#2774] [Q&A] [answered]
I was trying to write something like this:
```cpp
const json& foo = data["foo"];
if (!foo.is_null())
{
  ...
}
```
Where ` data ` would also be ` const json& `

This works well for non-const objects, but in this case it leads to abort. Is there any good solution for that?

**Comment by nlohmann** [comment, marked as answer]
Use at() or contains().

**Reply by Wokarol** [reply]
But using .contains() requires basically two searches in the json file and .at() requires exceptions. Both of which aren't ideal

**Reply by nlohmann** [reply]
See https://json.nlohmann.me/features/element_access/unchecked_access/ for the rationale. Alternatively, use find().

**Reply by jjcasmar** [reply]
I sometimes use `data.value("foo", json());` and can check for null as you do

**Reply by nlohmann** [reply]
You could also use [`find`](https://json.nlohmann.me/api/basic_json/find/) and check the returned iterator.
